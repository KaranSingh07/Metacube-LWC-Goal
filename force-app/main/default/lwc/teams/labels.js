export const LABELS = {
	ToastErrorTitle: 'Error!',
};
