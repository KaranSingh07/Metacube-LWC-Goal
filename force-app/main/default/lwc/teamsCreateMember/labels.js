export const LABELS = {
	ToastSuccessTitle: 'Success!',
	ToastErrorTitle: 'Error!',
	TeamMemberCreated: 'Team member created successfully.',
};
