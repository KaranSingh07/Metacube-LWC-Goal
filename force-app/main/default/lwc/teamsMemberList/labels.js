export const LABELS = {
	ToastSuccessTitle: 'Success!',
	ToastErrorTitle: 'Error!',
	TeamMembersSynced: 'Team members synced successfully.',
	NoTeamSelectedWarning: 'Select a team to show their members.',
	NoTeamMembersExistsWarning: 'This team have no members. Create one from above.',
};
