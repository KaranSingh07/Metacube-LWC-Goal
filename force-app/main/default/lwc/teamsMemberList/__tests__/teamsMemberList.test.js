import TeamsMemberList from 'c/teamsMemberList';
import { createElement } from 'lwc';
import { axe, toHaveNoViolations } from 'jest-axe';
import { setImmediate } from 'timers';
import { LABELS } from '../labels';
import getAllTeamMembers from '@salesforce/apex/TeamController.getAllTeamMembers';

jest.mock(
	'@salesforce/apex/TeamController.getAllTeamMembers',
	() => {
		return {
			default: jest.fn(),
		};
	},
	{ virtual: true }
);

expect.extend(toHaveNoViolations);

const mockTeams = require('./data/teams.json');
const mockTeamMembers = require('./data/teamMembers.json');
const LIGHTNING_SHOWTOAST_EVENT_NAME = 'lightning__showtoast';
const flushPromises = () => new Promise((resolve) => setImmediate(resolve));

const createComponent = () => {
	return createElement('c-teams-member-list', {
		is: TeamsMemberList,
	});
};

describe('TeamsMemberList Component', () => {
	afterEach(() => {
		while (document.body.firstChild) {
			document.body.removeChild(document.body.firstChild);
		}
		jest.clearAllMocks();
	});

	it('Should be accessible', async () => {
		// Given
		getAllTeamMembers.mockResolvedValue(mockTeamMembers);
		const component = createComponent();
		component.teams = mockTeams;

		// When
		document.body.appendChild(component);

		// Then
		expect(await axe(component)).toHaveNoViolations();
	});

	it('Should populate teams in memberTeam combobox when component is loaded', async () => {
		// Given
		getAllTeamMembers.mockResolvedValue(mockTeamMembers);
		const component = createComponent();
		component.teams = mockTeams;

		// When
		document.body.appendChild(component);
		await flushPromises();

		// Then
		const memberTeamCombobox = component.shadowRoot.querySelector(
			'lightning-combobox[data-id="memberTeam"]'
		);
		expect(memberTeamCombobox.options).toStrictEqual(mockTeams);
	});

	it('Should show correct default warning message when no team is selected, should not show team members', async () => {
		// Given
		getAllTeamMembers.mockResolvedValue(mockTeamMembers);
		const component = createComponent();
		component.teams = mockTeams;

		// When
		document.body.appendChild(component);
		await flushPromises();

		// Then
		const warningMessageComponent = component.shadowRoot.querySelector(
			'label[data-id="warningMessage"]'
		);
		const teamMembers = component.shadowRoot.querySelectorAll('div[data-id="teamMember"]');

		expect(getAllTeamMembers).toHaveBeenCalledTimes(1);
		expect(warningMessageComponent.innerHTML).toBe(LABELS.NoTeamSelectedWarning);
		expect(teamMembers.length).toBe(0);
	});

	it('Should show correct warning message when a team is selected that do not have any members, should not show team members', async () => {
		// Given
		getAllTeamMembers.mockResolvedValue(mockTeamMembers);
		const component = createComponent();
		component.teams = mockTeams;

		document.body.appendChild(component);
		await flushPromises();

		// When
		const selectTeamCombobox = component.shadowRoot.querySelector(
			'lightning-combobox[data-id="memberTeam"]'
		);
		selectTeamCombobox.dispatchEvent(new CustomEvent('change', { detail: { value: 'aus' } }));
		await flushPromises();

		// Then
		const warningMessageComponent = component.shadowRoot.querySelector(
			'label[data-id="warningMessage"]'
		);
		const teamMembers = component.shadowRoot.querySelectorAll('div[data-id="teamMember"]');

		expect(getAllTeamMembers).toHaveBeenCalledTimes(1);
		expect(warningMessageComponent.innerHTML).toBe(LABELS.NoTeamMembersExistsWarning);
		expect(teamMembers.length).toBe(0);
	});

});
