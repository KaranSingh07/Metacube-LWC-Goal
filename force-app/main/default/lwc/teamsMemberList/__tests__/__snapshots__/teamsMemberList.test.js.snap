// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamsMemberList Component Should show correct team members when a team is selected that have members, should not show warning message 1`] = `
StaticNodeList {
  "0": <div
    class="tile"
    data-id="teamMember"
  >
    <p>
      Sachin Tendulkar
    </p>
    <p>
      India
    </p>
    <p>
      Batting
    </p>
  </div>,
  "1": <div
    class="tile"
    data-id="teamMember"
  >
    <p>
      Virat Kohli
    </p>
    <p>
      India
    </p>
    <p>
      Batting
    </p>
  </div>,
  "2": <div
    class="tile"
    data-id="teamMember"
  >
    <p>
      Ravindra Jadeja
    </p>
    <p>
      India
    </p>
    <p>
      Batting, Bowling
    </p>
  </div>,
}
`;
